torch_geometric/profile/profile.py

```diff
@@ -92,36 +95,32 @@ def wrapper(*args, **kwargs) -> Tuple[Any, Stats]:
     return decorator
 
 
-def timeit():
-    r"""A decorator to facilitate timing a function, *e.g.*, obtaining the
-    runtime of a specific model on a specific dataset.
+class timeit(ContextDecorator):
+    r"""A ContextDecorator to facilitate timing a function, *e.g.*, obtaining
+    the runtime of a specific model on a specific dataset.
 
     .. code-block:: python
 
-        @timeit()
         @torch.no_grad()
         def test(model, x, edge_index):
             return model(x, edge_index)
 
-        z, time = test(model, x, edge_index)
+        with timeit() as timeit:
+            z = test(model, x, edge_index)
+        time = timeit.duration
```

Comment:
```suggestion
        with timeit() as t:
            z = test(model, x, edge_index)
        time = t.duration
```
otherwise, we are overriding `timeit` I assume.

Reply:
Done.